tests/test_store/test_fsspec.py

```diff
@@ -244,3 +258,27 @@ def test_no_wrap_async_filesystem():
 
     assert not isinstance(store.fs, AsyncFileSystemWrapper)
     assert store.fs.async_impl
+
+
+@pytest.mark.skipif(
+    parse_version(fsspec.__version__) < parse_version("2024.12.0"),
+    reason="No AsyncFileSystemWrapper",
+)
+async def test_delete_dir_wrapped_filesystem(tmpdir) -> None:
+    """The local fs is not async so we should expect it to be wrapped automatically"""
+    from fsspec.implementations.asyn_wrapper import AsyncFileSystemWrapper
+
+    store = FsspecStore.from_url(tmpdir / "test/path", storage_options={"auto_mkdir": True})
+
+    assert isinstance(store.fs, AsyncFileSystemWrapper)
+    assert store.fs.async_impl
+
```

Comment:
and `store.fs.asynchronous` ?

Reply:
I realize now you probably mean assert both `store.fs.asynchronous` and `store.fs.async_impl` ? I think I had misunderstood and replaced it.

Reply:
Both is good, although the isinstance line above should already make async_impl a certainty - so replacement is good too.